learn/src/main/java/org/navgurukul/learn/courses/network/RateLimiter.kt

```diff
@@ -0,0 +1,35 @@
+package org.navgurukul.learn.courses.network
+
+import android.os.SystemClock
+import androidx.collection.ArrayMap
+import java.util.concurrent.TimeUnit
+
+
+class RateLimiter<String>(timeout: Int, timeUnit: TimeUnit) {
+    private val timestamps = ArrayMap<String, Long>()
```

Comment:
If this variable is local, wont it be cleared when app is relaunched ? So if I launch and relaunch the app in quick succession, it will make the API call again ? please correct me if I am wrong.

Reply:
Deleted unused class